### ISSUE TITLE
iframe handling

### ISSUE THREAD

**Issue by bayfall** [stirno/FluentAutomation#49]
Need ability to switch document in case of iframes in page. Or have a way of having access to the driver and do it classic way ie 

driver.SwitchTo().Frame(someFrame).

Maybe this can be achieved in another way already , but I couldn't find it.

Thanks for your help

**Comment by stirno** [comment]
This has been on my 'gotta get it done' list for awhile now. I'll get it prioritized for the official v2 release.

**Comment by bayfall** [comment]
Cheers :-)

**Comment by 0liver** [comment]
@bayfall You can use the approach I mentioned in [a comment to a different issue](https://github.com/stirno/FluentAutomation/issues/66#issuecomment-11133866) to access the underlying `IWebDriver` instance, on which you'll be able to call `.SwitchTo()` etc. Of course, only if you're using Selenium ;-)

**Comment by stirno** [comment]
Implemented/committed for v3.0

Examples:

```
I.Switch.Frame("iframe:eq(0)"); // switch to first iframe
I.Switch.Frame("this_is_my_iframe_name"); // switch to iframe by name attribute
I.Switch.Frame("traditional_frame_name"); // switch to frame by name attribute
I.Switch.Frame(); // restore back to top level document

I.Switch.Window("My Application"); // switch window by title
I.Switch.Window("/contact"); // switch window by URL (substring match, EndsWith basically)
I.Switch.Window(); // switch back to the default window
```

**Comment by stirno** [comment]
Confirmed working as best as we can right now in the 3.0 branch (fixmultibrowsertesting -- stupid name I know..). Should be up on NuGet in a few days.
